src/BlockDecomposition.jl

```diff
@@ -43,14 +50,18 @@ function model_factory(::Val{false}, args...; kw...)::JuMP.Model
     return JuMP.Model(args...; kw...)
 end
 
-function BlockModel(args...; kw...)
+function BlockModel(args...; automatic_dantzig_wolfe::AutoDwStrategy = inaktive, kw...)
     dm = haskey(kw, :direct_model) ? kw[:direct_model] : false
     m = model_factory(Val(dm), args...; kw...)
     JuMP.set_optimize_hook(m, optimize!)
+    m.ext[:automatic_dantzig_wolfe] = automatic_dantzig_wolfe
```

Comment:
Yes, in a similar manner an automatic Benders decomposition can be implemented. However, a different structure detection algorithm is needed because so far only structures with no linking variables are detected. Also, the scores are explicitly constructed to score structures according to their suitability to a Dantzig-Wolfe decomposition. Maybe we could rename it to `automatic_dantzig_wolfe_strategy`? In case we rename it to `automatic_decomposition_strategy` it might be a good idea to rename the strategies (so e.g. `dw_white_score` instead of `white_score`). If an automatic Benders decomposition will then be implemented at some point of time, the scores used for it could be named `bd_....` and can thereby distinguished from the Dantzig-Wolfe scores named `dw_...`.